compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -1254,23 +1246,15 @@ object Semantic:
       iterate()
     end doTask
   end WorkList
-  inline def workList(using wl: WorkList): WorkList = wl
 
 // ----- API --------------------------------
 
-  /** Add a checking task to the work list */
-  def addTask(thisRef: ThisRef)(using WorkList) = workList.addTask(Task(thisRef))
-
-  /** Check the specified tasks
-   *
-   *      Semantic.checkTasks {
-   *         Semantic.addTask(...)
-   *      }
+  /**
+   * Check the specified concrete classes
    */
-  def checkTasks(using Context)(taskBuilder: WorkList ?=> Unit): Unit =
-    val workList = new WorkList
+  def checkClasses(concreteClasses: List[ClassSymbol])(using Context): Unit =
+    val workList = new WorkList(concreteClasses)
```

Comment:
I was surprised that the worklist was already not really a worklist, in that no new classes get added to it while it is processing existing classes. Given that this is the case, this simplification makes sense (i.e. getting rid of the mutable `pendingTasks`). But it would make sense to take the simplification a step further and get rid of the Worklist class altogether, keeping only its doTask method. Then here we would just do `concreteClasses.filter(isConcreteClass).foreach(doTask)`. Also we should rename `concreteClasses` to just `classes` or `classesToCheck` or `allClasses` or ...